saged/utils.py

```diff
@@ -169,3 +172,111 @@ def run_limma(expression_values: np.array,
         return limma.removeBatchEffect(expression_values, batches)
     else:
         return limma.removeBatchEffect(expression_values, batches, second_batch)
+
+
+def parse_label_file(label_file_path: Union[str, Path]) -> Dict[str, str]:
+    """
+    Create a sample to label mapping from the pickled file output by label_samples.py
+
+    Arguments
+    ---------
+    map_file_path: The path to a pickled file created by label_samples.py
+
+    Returns
+    -------
+    sample_to_label: A string to string dict mapping sample ids to their
+        corresponding label string. E.g. {'GSM297791': 'sepsis'}
+    """
+    sample_to_label = {}
+    label_to_sample = None
+    with open(label_file_path, 'rb') as map_file:
+        label_to_sample, _ = pickle.load(map_file)
+
+    for label in label_to_sample:
+        for sample in label_to_sample[label]:
```

Comment:
The current version of the labeling script I used only allows one label per sample. That being said, it could change in the future so you're right that I should add an assert statement to make sure that is true